src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2115,7 +2117,15 @@ impl Coordinator {
             }
         };
         let source_ids = source.depends_on();
-        let timeline_context = self.validate_timeline_context(source_ids.clone())?;
+        let mut timeline_context = self.validate_timeline_context(source_ids.clone())?;
+        if matches!(timeline_context, TimelineContext::TimestampIndependent)
+            && source.contains_temporal()
+        {
+            // If the source IDs are timestamp independent but the query contains temporal functions,
+            // then the timeline context needs to be upgraded to timestamp dependent. This is
+            // required because `source_ids` doesn't contain functions.
```

Comment:
> This is required because `source_ids` doesn't contain functions.

I know this is copy-pasted, but I don't really understand this sentence. Any idea what it means?

Reply:
I guess in an alternate universe, `source_ids` would involve function ids, such as `mz_now()` (like how `resolved_ids` does), and then `validate_timeline_context` could detect `mz_now()` by itself.